opencog/atoms/bind/PatternLink.cc

```diff
@@ -506,14 +548,8 @@ void PatternLink::unbundle_virtual(const std::set<Handle>& vars,
 		{
 			_pat.evaluatable_terms.insert(sh);
 			add_to_map(_pat.in_evaluatable, sh, sh);
-			// But they're virtual only if they have two or more
-			// unquoted, bound variables in them. Otherwise, they
-			// can be evaluated on the spot.
```

Comment:
Removing this comment is wrong, and makig everything virtual, below, is wrong.    Consider, for example, `GreaterThanLink`, and we wish to evaluate whether it is true or not.  There are three cases:

`if 1 < 2` -- this can be evaluated immediately, there are no variables at all.

`if 1 < x` -- this can be evaluated as soon as a grounding for variable x is found.  You do not need any of the complicated virtual-link code to determine if this is true or not; you only need the grounding for x.

`if x < y` -- this is a true "virtual link" that requires the _temp_atomspace and the other complicated machinery.  The problem is that every different possible grounding for x must be compared to every different possible grounding for y.   Terms such as this cannot be evaluated immediately; they must be saved for last, only after each different possible x and y have been determined.  Only this last form are really, truly "virtual", in that they require the complex virtual-code to evaluate them.